app/feature/feature-payments/src/main/kotlin/com/hedvig/android/feature/payments2/PaymentOverviewDestination.kt

```diff
@@ -0,0 +1,284 @@
+            .padding(16.dp),
+        )
+        Divider(modifier = Modifier.padding(horizontal = 16.dp))
+        PaymentHistory(
+          modifier = Modifier
+            .clickable { onPaymentHistoryClicked() }
+            .padding(16.dp),
+        )
+        Divider(modifier = Modifier.padding(horizontal = 16.dp))
+        PaymentDetails(
+          displayName = uiState.connectedPaymentDisplayName,
+          displayValue = uiState.connectedPaymentValue,
+        )
+        Spacer(Modifier.height(16.dp))
+        if (uiState.hasConnectedPayment) {
+          HedvigContainedButton(
+            text = stringResource(R.string.PROFILE_PAYMENT_CHANGE_BANK_ACCOUNT),
+            onClick = onChangeBankAccount,
+            modifier = Modifier.padding(horizontal = 16.dp),
+          )
+        } else {
+          HedvigContainedButton(
+            text = stringResource(R.string.PROFILE_PAYMENT_CONNECT_DIRECT_DEBIT_BUTTON),
+            onClick = onChangeBankAccount,
+            modifier = Modifier.padding(horizontal = 16.dp),
+          )
+        }
+        Spacer(Modifier.height(16.dp))
+      }
+
+      is OverViewUiState.Error -> Box(Modifier.weight(1f), contentAlignment = Alignment.Center) {
+        HedvigErrorSection(retry = onRetry)
+      }
+
+      OverViewUiState.Loading -> HedvigFullScreenCenterAlignedProgressDebounced()
+    }
+  }
+}
+
+@Composable
+private fun PaymentAmountCard(
+  uiState: OverViewUiState.Content,
+  onUpcomingPaymentClicked: () -> Unit,
+  modifier: Modifier = Modifier,
+) {
+  HedvigCard(
+    onClick = onUpcomingPaymentClicked,
+    modifier = modifier,
+  ) {
+    Column(
+      modifier = Modifier
+        .padding(horizontal = 16.dp, vertical = 12.dp)
+        .fillMaxWidth(),
+    ) {
+      HorizontalItemsWithMaximumSpaceTaken(
+        startSlot = { Text(stringResource(id = R.string.PAYMENTS_UPCOMING_PAYMENT)) },
+        endSlot = {
+          Row(
+            Modifier.fillMaxWidth(),
+            horizontalArrangement = Arrangement.End,
+            verticalAlignment = Alignment.CenterVertically,
+          ) {
+            Text(
+              text = uiState.upcomingPayment.toString(),
+              textAlign = TextAlign.End,
+            )
+            Spacer(Modifier.width(4.dp))
+            Icon(
+              imageVector = Icons.Hedvig.ChevronRight,
+              contentDescription = null,
+              tint = MaterialTheme.colorScheme.onSurfaceVariant,
+              modifier = Modifier.size(16.dp),
+            )
+          }
+        },
+      )
+      val dateTimeFormatter = rememberHedvigDateTimeFormatter()
+      Text(
+        text = stringResource(
+          id = R.string.GENERAL_DUE,
+          dateTimeFormatter.format(uiState.dueDate.toJavaLocalDate()),
+        ),
+        color = MaterialTheme.colorScheme.onSurfaceVariant,
+      )
+    }
+  }
+}
+
+@Suppress("UnusedReceiverParameter")
+@Composable
+private fun ColumnScope.PaymentDetails(
```

Comment:
No need for this to be extension on `ColumnScope` if it does not need any modifier which is only applicable under `ColumnScope` and if it does not emit more than 1 composable top-level (here it just emits one Row composable)